source/views/Extension/Views/ImportWallet/Steps/Step2.jsx

```diff
@@ -36,13 +30,44 @@ const Step2 = () => {
     }
   }, [currentEmoji, walletName]);
 
+  const onEmojiClick = (_event, emojiObject) => {
+    setCurrentEmoji(emojiObject.emoji);
+    setOpenEmojis(false);
+    setOpenEmojiSelector(false);
+  };
+
+  const createImportedAccount = (pemContent) => {
+    console.log(pemContent);
+    sendMessage(
+      {
+        type: HANDLER_TYPES.IMPORT_PEM_ACCOUNT,
+        params: { icon: currentEmoji, name: walletName, pem: pemContent },
+      },
+      (wallet) => {
+        console.log(wallet);
+      }
+    );
+    setLoading(false);
+    navigator.navigate("home");
+  };
+
+  const readPemCreateAccount = () => {
+    setLoading(true);
+    setDisabled(true);
+    let fileReader = new FileReader();
+    fileReader.readAsText(userPemFile);
+    fileReader.onloadend = () => {
+      const content = fileReader.result; // we should check what happens if the user pass an empty pem file
```

Comment:
has this been checked? Honestly if that's the case, serialization should fail on the BE right @dantefarias ?

Reply:
If the pem file is invalid or empty BE is gonna throw an error about **Invalid Key**